docs/problem-matchers.md

```diff
@@ -0,0 +1,116 @@
+# Problem Matchers
+Problem Matchers are a way to scan the output of actions for a specified regex pattern and surface that information prominently in the UI. Both [GitHub Annotations](https://developer.github.com/v3/checks/runs/#annotations-object-1) and log file decorations are created when a match is detected.
+
+## Single Line Matchers
+
+Let's consider the ESLint compact output:
+```
+badFile.js: line 50, col 11, Error - 'myVar' is defined but never used. (no-unused-vars)
+```
+We can define a problem matcher in json that detects input in that format:
+```
+{
+    "problemMatcher": [
+        {
+            "owner": "eslint-compact",
+            "pattern": [
+                {
+                    // The pattern that provides groups to match against
```

Comment:
comments are not legal JSON

Reply:
we shouldnt suggest customers use them, if it we switch to a different json parser in the future, it might break.